prow/plugins/blunderbuss/blunderbuss.go

```diff
@@ -289,11 +302,14 @@ func getReviewers(rc reviewersClient, ghc githubClient, log *logrus.Entry, autho
 		fileReviewers := rc.Reviewers(file.Filename).Difference(authorSet)
 		for reviewers.Len() < minReviewers && fileReviewers.Len() > 0 {
 			if r := findReviewer(ghc, log, useStatusAvailability, &busyReviewers, &fileReviewers); r != "" {
-				reviewers.Insert(r)
+				if !reviewers.Has(r) {
+					reviewersList = append(reviewersList, r)
+					reviewers.Insert(r)
+				}
 			}
 		}
 	}
-	return reviewers.List(), requiredReviewers.List(), nil
+	return reviewersList, requiredReviewers.List(), nil
```

Comment:
This seems like an unnecessary (and worse) change from the existing code

Reply:
Sorry Erick but this is actually what makes this fix working... because the various loops fill reviewers in a specific order, from the leaf first up to the root.
By using the `sets.String` we are losing this information and then when we truncate the list to `maxReviewerCount` Ben goes first...

Reply:
I have decided to keep the `reviewers` set, because it is used in all the logic with unions and co.

Reply:
Why do we depend on the order? This seems very wrong.

Reply:
Because the list of reviewers is constructed this way to give priority to leaf approvers when there's a need to shorten the list.
You can revert that line and see how the new test fails in suggesting Ben...

Reply:
@fejta how would you do it?

Reply:
Ordering the string seems like it will prefer Ben to Erick, when we want each person in a given owners file assigned at roughly the same frequency.

I would use a stack of sets, one per level.

AKA if the file is:

`alpha/beta/omega/something/OWNERS`

and there are 

`alpha/OWNERS`, `alpha/beta/OWNERS` and `alpha/beta/omega/something/OWNERS`

I would represent this as: `[]sets.String`

I would choose someone by something along the lines of:

```
for i := range options {
  all := options[len(options)-i] // start from leaf/OWNERS not /OWNERS
  rand.Shuffle(all, func(i,j int) { all[j], all[i] = all[i], all[j] })
  for _, choice := range all {
    if chosen.Contains(choice) {
      continue
    }
    if enoughChoices(chosen) { break/return/whatever }
  }
  log.Warn("too few choices, oh well")
}
```

Reply:
Ok, thanks for having taken the time to look deeply into it. I will implement that during the rainy weekend...